platform/darwin/src/MGLRendererConfiguration.mm

```diff
@@ -1,19 +1,62 @@
 #import "MGLRendererConfiguration.h"
 #import "MGLOfflineStorage_Private.h"
+#import "MGLFoundation_Private.h"
 
 #if TARGET_OS_IPHONE
 #import <UIKit/UIKit.h>
 #else
 #import <AppKit/AppKit.h>
 #endif
 
+static NSString * const MGLCollisionBehaviorPre4_0Key = @"MGLCollisionBehaviorPre4_0";
+
+@interface MGLRendererConfiguration ()
+@property (nonatomic, readwrite) BOOL perSourceCollisions;
+@end
+
 
 @implementation MGLRendererConfiguration
 
 + (instancetype)currentConfiguration {
     return [[self alloc] init];
 }
 
+- (instancetype)init {
+    return [self initWithPropertyDictionary:[[NSBundle mainBundle] infoDictionary]];
+}
+
+- (instancetype)initWithPropertyDictionary:(nonnull NSDictionary *)properties {
+    self = [super init];
+    
+    if (self) {
+        // Set the collision behaviour. A value set in `NSUserDefaults.standardUserDefaults`
+        // should override anything in the application's info.plist
+        NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+
+        if ([defaults objectForKey:MGLCollisionBehaviorPre4_0Key]) {
+            _perSourceCollisions = [defaults boolForKey:MGLCollisionBehaviorPre4_0Key];
+        }
+        else {
+            id collisionBehaviourValue = properties[MGLCollisionBehaviorPre4_0Key];
+            
+            NSNumber *collisionBehaviourNumber = MGL_OBJC_DYNAMIC_CAST(collisionBehaviourValue, NSNumber);
+            
+            if (collisionBehaviourNumber) {
+                _perSourceCollisions = collisionBehaviourNumber.boolValue;
+            } else {
+                // Also support NSString to correspond with the behavior of `-[NSUserDefaults boolForKey:]`
```

Comment:
I don't think this is necessary because `boolForKey` coerces "YES"/YES/1/"1"/"NO"/NO/0/"0" as `BOOL`

Reply:
That's only on `NSUserDefaults` unfortunately - this is for the property from the `NSDictionary` (from the bundle/info.plist). So, I think we need this? 

Though strictly speaking it's not documented that `@"YES"`/`@"NO"` etc. are supported, but given that this PR introduces support for `NSUserDefaults` I figured this ought to be included.

(An `mgl_boolForKey` might be a nice category method to add to `NSDictionary`)